src/ILCompiler.Compiler/src/Compiler/MethodExtensions.cs

```diff
@@ -77,13 +77,11 @@ public static SpecialMethodKind DetectSpecialMethodKind(this MethodDesc method)
         {
             if (method.IsPInvoke)
             {
-                // Marshalling is never required for pregenerated interop code
-                if (Internal.IL.McgInteropSupport.IsPregeneratedInterop(method))
-                {
-                    return SpecialMethodKind.PInvoke;
-                }
-
-                if (!Internal.IL.Stubs.PInvokeMarshallingILEmitter.RequiresMarshalling(method))
+                // Only treat the synthetic target methods as real PInvoke.
```

Comment:
The pre-generated interop should still go directly here, without going through `PInvokeMarshallingILEmitter`. We should encapsulate it in helper method - something like:

```
static bool IsRawPInvoke(MethodDesc method)
{
   return method.IsPInvoke && ((method is Internal.IL.Stubs.PInvokeTargetNativeMethod) || Internal.IL.McgInteropSupport.IsPregeneratedInterop(method));
}
```